Source/Core/Duality/Backend/DefaultPluginLoader.cs

```diff
@@ -99,42 +100,53 @@ public void Init()
 			// and more reliably.
 			{
 				string osName = Environment.OSVersion != null ? Environment.OSVersion.ToString() : "Unknown";
-				string osFriendlyName = null;
-				if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+				osName = "(" + osName + ")";
+
+				string osFriendlyName = "Other OS";
+
+				switch (Environment.OSVersion.Platform)
 				{
-					if (Environment.OSVersion.Version >= new Version(10, 0, 0))
-						osFriendlyName = "Windows 10";
-					else if (Environment.OSVersion.Version >= new Version(6, 3, 0))
-						osFriendlyName = "Windows 8.1";
-					else if (Environment.OSVersion.Version >= new Version(6, 2, 0))
-						osFriendlyName = "Windows 8";
-					else if (Environment.OSVersion.Version >= new Version(6, 1, 0))
-						osFriendlyName = "Windows 7";
-					else if (Environment.OSVersion.Version >= new Version(6, 0, 0))
-						osFriendlyName = "Windows Vista";
-					else if (Environment.OSVersion.Version >= new Version(5, 2, 0))
-						osFriendlyName = "Windows XP 64 Bit Edition";
-					else if (Environment.OSVersion.Version >= new Version(5, 1, 0))
-						osFriendlyName = "Windows XP";
-					else if (Environment.OSVersion.Version >= new Version(5, 0, 0))
-						osFriendlyName = "Windows 2000";
+					case PlatformID.Win32NT:
+						if (Environment.OSVersion.Version >= new Version(6, 2, 0))
+							osFriendlyName = "Windows 8 or higher";
+						else if (Environment.OSVersion.Version >= new Version(6, 1, 0))
+							osFriendlyName = "Windows 7";
+						else if (Environment.OSVersion.Version >= new Version(6, 0, 0))
+							osFriendlyName = "Windows Vista";
+						else if (Environment.OSVersion.Version >= new Version(5, 2, 0))
+							osFriendlyName = "Windows XP 64 Bit Edition";
+						else if (Environment.OSVersion.Version >= new Version(5, 1, 0))
+							osFriendlyName = "Windows XP";
+						else if (Environment.OSVersion.Version >= new Version(5, 0, 0))
+							osFriendlyName = "Windows 2000";
+						break;
+
+					case PlatformID.MacOSX:
+						osFriendlyName = "OSX";
+						break;
+
+					case PlatformID.Unix:
+						osFriendlyName = "Generic UNIX";
+						break;
 				}
-				Log.Core.Write(
-					"Environment Info: " + Environment.NewLine +
-					"  Current Directory: {0}" + Environment.NewLine +
-					"  Command Line: {1}" + Environment.NewLine +
-					"  Operating System: {2}" + Environment.NewLine +
-					"  64 Bit OS: {3}" + Environment.NewLine +
-					"  64 Bit Process: {4}" + Environment.NewLine +
-					"  CLR Version: {5}" + Environment.NewLine +
-					"  Processor Count: {6}",
-					Environment.CurrentDirectory,
-					Environment.CommandLine,
-					osName + (osFriendlyName != null ? (" (" + osFriendlyName + ")") : ""),
-					Environment.Is64BitOperatingSystem,
-					Environment.Is64BitProcess,
-					Environment.Version,
-					Environment.ProcessorCount);
+
+				StringBuilder envInfo = new StringBuilder();
+				envInfo.AppendLine("Environment Info:")
```

Comment:
- 👍 for refactoring to use `StringBuilder` in this case.
- Please don't use fluent syntax unless there's a very good reason to.
- Consider (not sure, you decide) using `AppendFormat(...)` + `AppendLine()` combo instead of repeated `string.Format`, might otherwise allocate as much as previous version without `StringBuilder`.